### DISCUSSION TITLE
Problem saving graphics to file

### DISCUSSION THREAD

**Discussion by kezhaozhang** [WLJSTeam/wljs-notebook#210] [General]
I used the Export["plot.jpg", %] command to savee, but the resulting file only contains the axes. Is there a way to properly export graphics to files? Thanks.

**Comment by JerryI** [comment]
Yea, sure. Since it is rendered using frontend (not Kernel like in Mathematica), you need to locate a right top corner of your plot. There will be a dropdown menu for that.

**Reply by kezhaozhang** [reply]
Thanks. It works! I didn't realize the dropdown menu existed.

**Comment by kezhaozhang** [comment]
One more question: It appears that plot legends are not saved. Is there a way to save them as well?

**Reply by JerryI** [reply]
Unfortunately, Mathematica originally places legends as a wrapper around any arbitrary expression. In this manner, `Graphics` doesn't know about legends :( 

I found 4 options 

1. Take a screenshot (kinda lame...)
2. Build legend function by your own using `Epilog` option with `Rectangle`, `Text`... Or make it on a separate graph and place them onto your plot using `Inset`
3. Use alternative library for plotting such as built-in `Plotly` (see [docs](https://jerryi.github.io/wljs-docs/frontend/Reference/Plotly/))
4. Export as HTML file an entire notebook (click on `Share` button in the top bar). It might not be suitable for all purposes, more for sharing a notebook with other people. Code cell you can be hidden using arrow button located on the left-bottom corner of input-output pair.

**Reply by kezhaozhang** [reply]
Thanks for the detailed and helpful information! I'll take a look at Plotly.

**Reply by JerryI** [reply]
Finally fixed!
